build_miniforge_osx.sh

```diff
@@ -4,14 +4,15 @@ set -e
 set -x
 
 echo "Installing a fresh version of Miniforge3."
-MINIFORGE_URL="https://github.com/conda-forge/miniforge/releases/download/4.8.3-1"
-MINIFORGE_FILE="Miniforge3-4.8.3-1-MacOSX-x86_64.sh"
+MINIFORGE_VERSION="4.10.3-8"
+MINIFORGE_URL="https://github.com/conda-forge/miniforge/releases/download/${MINIFORGE_VERSION}"
+MINIFORGE_FILE="Mambaforge-${MINIFORGE_VERSION}-MacOSX-x86_64.sh"
 curl -L -O "${MINIFORGE_URL}/${MINIFORGE_FILE}"
 bash "${MINIFORGE_FILE}" -b
```

Comment:
Should we set the path with `-p` here so that paths below don't need to be updated?

Reply:
I'm not sure how the `-p` flag helps here. Can you elaborate? Will it avoid the command

```
source ~/mambaforge/bin/activate
```
or will it make it explicit that we are installing it at the directory `~/mambaforge`

Reply:
It could be used to call this something generic like `conda_install`. Admittedly somewhat cosmetic, but it also saves us changing another line if we change this again in the future